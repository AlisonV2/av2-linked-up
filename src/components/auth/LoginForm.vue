<template>
  <form class="auth-form" @submit.prevent="login">
    <div class="row">
      <app-title mode="title-img">Welcome back!</app-title>
    </div>
    <div class="form-floating mb-3">
      <input
        type="email"
        class="form-control"
        placeholder="name@example.com"
        v-model="email"
        required
      />
      <label>Email</label>
    </div>
    <div class="form-floating mb-3">
      <input
        type="password"
        class="form-control"
        placeholder="Password"
        v-model="password"
        required
      />
      <label>Password</label>
    </div>
    <div class="error mb-3 mt-3">{{ error }}</div>
    <div class="btn-center">
      <app-button>Log in</app-button>
    </div>
  </form>
</template>

<script>
/**
 * @type Component
 * @name LoginForm
 * @vue-data {string} email - v-model
 * @vue-data {string} password - v-model
 * @vue-data {string} error
 * @vue-event login
 */
export default {
  data() {
    return {
      email: '',
      password: '',
      error: null,
    };
  },
  methods: {
    /**
     * @description Dispatch login store action with form data
     * Redirect to AdminProfile on success
     * @method login
     * @param {object} userData
     * @async
     */
    async login() {
      const userData = {
        email: this.email,
        password: this.password,
      };
      try {
        await this.$store.dispatch('login', userData);
      } catch (err) {
        this.error = err.message;
        return;
      }
      this.$router.push({ name: 'AdminProfile' });
    },
  },
};
</script>
