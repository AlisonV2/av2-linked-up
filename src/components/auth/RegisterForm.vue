<template>
  <form class="auth-form" @submit.prevent="register">
    <div class="row">
      <app-title>Glad you join us!</app-title>
    </div>
    <div class="form-floating mb-3">
      <input
        type="text"
        class="form-control"
        placeholder="Your name"
        v-model="name"
        required
      />
      <label>Public Name</label>
    </div>
    <div class="form-floating mb-3">
      <input
        type="email"
        class="form-control"
        placeholder="name@example.com"
        v-model="email"
        required
      />
      <label>Email</label>
    </div>
    <div class="form-floating mb-3">
      <input
        type="password"
        class="form-control"
        placeholder="Password"
        v-model="password"
        required
      />
      <label>Password</label>
    </div>
    <div>
      <select class="form-select" v-model="role" required>
        <option disabled value="">You are...</option>
        <option value="artist">An artist</option>
        <option value="client">A (future) client</option>
        <!-- <option value="organizer">An organizer</option> -->
        <!-- <option value="admin">Admin</option> -->
      </select>
    </div>
    <div class="error mb-3 mt-3">{{ error }}</div>
    <div class="btn-center">
      <app-button mode="signup-btn">Sign up</app-button>
    </div>
  </form>
</template>

<script>
import * as Sentry from '@sentry/vue';

/**
 * @exports RegisterForm
 * @type {Component}
 * @requires Sentry
 * @vue-data {string} name - v-model
 * @vue-data {string} email - v-model
 * @vue-data {string} password - v-model
 * @vue-data {string} role - v-model
 * @vue-data {string} error
 * @vue-event register
 */
export default {
  name: 'RegisterForm',
  data() {
    return {
      name: '',
      email: '',
      password: '',
      role: '',
      error: null,
    };
  },
  methods: {
    /**
     * @description Dispatch register store action with form data
     * Redirect to AdminProfile on success
     * @method register
     * @param {object} userData
     * @async
     */
    async register() {
      const userData = {
        name: this.name,
        email: this.email,
        password: this.password,
        role: this.role,
      };
      try {
        await this.$store.dispatch('register', userData);
      } catch (err) {
        this.error = err.message;
        Sentry.captureException(err);
        return;
      }
      this.$router.push({ name: 'AdminProfile' });
    },
  },
};
</script>
