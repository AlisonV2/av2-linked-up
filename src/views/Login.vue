<template>
    <div class="container col-6 home">
    <div v-if="showLogin">
      <LoginForm />
      <p class="mt-3 text-center">
        No account yet? <span class="toggle-form show-signup" @click="showLogin = false">Sign up</span> instead
      </p>
    </div>
    <div v-else>
      <RegisterForm />
      <p class="mt-3 text-center">
        Already registered? <span class="toggle-form show-login" @click="showLogin = true">Log in</span> instead
      </p>
    </div>
  </div>
</template>

<script>
import RegisterForm from '@/components/auth/RegisterForm.vue';
import LoginForm from '@/components/auth/LoginForm.vue';

/**
 * @exports Login
 * @requires RegisterForm
 * @requires LoginForm
 * @type {Page}
 * @vue-data {boolean} showLogin - Toggles Login/Register form
 */
export default {
  name: 'Home',
  components: {
    RegisterForm,
    LoginForm
  },
  data() {
    return {
      showLogin: true
    }
  }
}
</script>

<style lang="scss">
.home {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 70vh;
}

.toggle-form {
    color: $primary;
    cursor: pointer;
}
</style>
