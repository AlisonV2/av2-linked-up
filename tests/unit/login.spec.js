import { shallowMount } from '@vue/test-utils';
import Login from '@/views/Login.vue';
import RegisterForm from '@/components/auth/RegisterForm.vue';
import LoginForm from '@/components/auth/LoginForm.vue';

const config = {
  data() {
    return {
      showLogin: true,
    };
  },
  global: {
    components: {
      RegisterForm,
      LoginForm,
    },
  },
};

/**
 * @module LoginPageTest
 */
describe('Login.vue', () => {
  it('Loginpage Snapshot', () => {
    const component = shallowMount(Login, config);
    expect(component.element).toMatchSnapshot();
  });

  it('Checks if LoginForm is rendered', () => {
    const component = shallowMount(Login, config);
    const loginForm = component.findComponent(LoginForm);
    expect(loginForm.exists()).toBe(true);
  });

  it('Checks if RegisterForm is not visible', () => {
    const component = shallowMount(Login, config);
    const registerForm = component.findComponent(RegisterForm);
    expect(registerForm.exists()).toBe(false);
  });

  it('Checks if LoginForm is not visible', () => {
    const component = shallowMount(Login, {
      ...config,
      data() {
        return {
          showLogin: false,
        };
      },
    });
    const loginForm = component.findComponent(LoginForm);
    expect(loginForm.exists()).toBe(false);
  });

  it('Checks if RegisterForm is visible', async () => {
    const component = shallowMount(Login, config);
    component.find('.show-signup').trigger('click');
    await component.vm.$nextTick();
    const registerForm = component.findComponent(RegisterForm);
    expect(registerForm.exists()).toBe(true);
  });
});
