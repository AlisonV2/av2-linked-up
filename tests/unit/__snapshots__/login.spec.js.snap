// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login.vue Loginpage Snapshot 1`] = `
<div
  class="container col-6 home"
>
  <div>
    <login-form-stub />
  </div>
</div>
`;
